src/NHibernate/Intercept/FieldInterceptionHelper.cs

```diff
@@ -32,6 +33,7 @@ public static IFieldInterceptor ExtractFieldInterceptor(object entity)
 			return fieldInterceptorAccessor == null ? null : fieldInterceptorAccessor.FieldInterceptor;
 		}
 
+		[Obsolete("Use IBytecodeEnhancementMetadata.InjectInterceptor method instead")]
```

Comment:
I think `IsInstrumented(object entity)` should be obsoleted too: all its remaining calls in NHibernate project can be replaced by `persister.IsInstrumented` calls, or simply removed. Two calls in tests would have to be replaced by an interface check.